fundingmanager.go

```diff
@@ -422,6 +426,23 @@ func (f *fundingManager) Start() error {
 
 		f.localDiscoverySignals[chanID] = make(chan struct{})
 
+		// Rebroadcast the funding transaction for any pending channel
+		// that we initiated. If this operation fails due to a reported
+		// double spend, we treat this as an indicator that we have
+		// already broadcast this transaction. Otherwise, we simply log
+		// the error as there isn't anything we can currently do to
+		// recover.
+		if channel.IsPending &&
+			channel.ChanType == channeldb.SingleFunder &&
```

Comment:
Not that this currently matters, but why not also broadcast in `DualFunder` mode?

Reply:
We could definitely do that as well, currently the logic to serialize the funding txn is only done under these conditions, so in dual funder mode the `FundingTxn` would be nil. Do we have a dual funder test in the funding manager?

Reply:
Thing is that `DualFunder` is not really accounted for or tested properly other places, so I think starting do check `SingleFunder` vs `DualFunder` would give the impression to someone reading the code that `DualFunder` was properly working. @Roasbeef will know this

Reply:
I think it's fine as `DualFunder` isn't exposed anywhere in the wire protocol. I've made effort to retain its functionality within `lnd` itself so when we switch over, there won't have to be large code disruption.